valueset.go

```diff
@@ -0,0 +1,179 @@
+package neo4j
+
+import (
+	"io"
+
+	"github.com/cloudprivacylabs/lpg"
+	"github.com/cloudprivacylabs/lsa/pkg/ls"
+)
+
+type NodesetInput interface {
+	ColumnNames() []string
+	Next() ([]string, error)
+	Reset() error
+}
+
+type Nodeset struct {
+	ID     string
+	Labels []string
+	Data   map[string]NodesetData
+	Rows   ItrRows
+}
+
+type ItrRows struct {
+	R    int
+	Rows [][]string
+}
+
+func (ir *ItrRows) next() ([]string, error) {
+	ir.R++
+	if ir.R >= len(ir.Rows) {
+		return nil, io.EOF
+	}
+	return ir.Rows[ir.R], nil
+}
+
+type NodesetData struct {
+	ID                 string
+	Labels             []string
+	Properties         map[string]interface{}
+	LinkedToOtherNodes bool
+}
+
+func (ns Nodeset) ColumnNames() []string {
+	cols := make([]string, 0)
+	for _, d := range ns.Data {
+		cols = append(cols, d.ID)
+	}
+	return cols
+}
+
+func (ns Nodeset) Next() ([]string, error) {
+	return ns.Rows.next()
+}
+
+func (ns Nodeset) Reset() error {
+
+}
+
+func ParseNodesetData(input NodesetInput) (map[string]Nodeset, error) {
+	processed := false
+	ret := make(map[string]Nodeset)
+	getColumn := func(row []string, header string) string {
+		for cIdx, c := range input.ColumnNames() {
+			if c == header {
+				return row[cIdx]
+			}
+		}
+		return ""
+	}
+	seenIDs := make(map[string]struct{})
+	for {
+		if err := input.Reset(); err != nil {
+			return nil, err
+		}
+		var prevNodesetID string
+		for row, err := input.Next(); ; {
+			if err == io.EOF {
+				seenIDs[prevNodesetID] = struct{}{}
+				break
+			}
+			if err != nil {
+				return nil, err
+			}
+			nodesetId := getColumn(row, "nodeset_id")
+			// Process a new nodeset
+			nodeset := Nodeset{
+				ID:   nodesetId,
+				Rows: ItrRows{Rows: make([][]string, 0)},
+			}
+			nodesetRows := nodeset.Rows.Rows
+			nodesetRows = append(nodesetRows, row)
+			// process a row
+			processed = true
+			for r, err := input.Next(); ; {
+				if err != nil {
```

Comment:
Here you need to check for eof. If eof, put the nodeset into ret map.
set currentNodesetId=""